### ISSUE TITLE
Problem with new events (fixed)

### ISSUE THREAD

**Issue by m49n** [mplodowski/formbuilder-plugin-public#45]
**Describe the bug**
it is impossible to change an object in an event

**Fix**
was:
event('formBuilder.overrideForm', [$this->form]);
became
event('formBuilder.overrideForm', [&$this->form]);

and Event::listen('formBuilder.overrideForm', function (&$form) with &

same in other events
was:
event('formBuilder.formSubmitted', [$this->form]);
became:
event('formBuilder.formSubmitted', [&$this->form]);

**Plugin version**
4.0.5

**Additional context**
Problem with new events

**Comment by mplodowski** [comment]
Hi,

please write me what you are trying to do, so I can confirm this.

For example, this code will override form css class property and it is rendered correctly without passing argument by reference:

`
Event::listen('formBuilder.overrideForm', function ($form) {
    $form->css_class = 'my_class';
});
`

**Comment by m49n** [comment]
> Hi,
> 
> please write me what you are trying to do, so I can confirm this.
> 
> For example, this code will override form css class property and it is rendered correctly without passing argument by reference:
> 
> ```
> Event::listen('formBuilder.overrideForm', function ($form) {
>     $form->css_class = 'my_class';
> });
> ```

for example I want to override the entire form
```
Event::listen('formBuilder.overrideForm', function (&$form) {
            if (request()->form) {
                $form = Form::query()
                    ->with([
                        'fields' => fn($query) => $query->isVisible()->with('field_type'),
                    ])
                    ->where('code', request()->form)
                    ->firstOrFail();
            }
        });
```

**Comment by m49n** [comment]
> **Describe the bug** it is impossible to change an object in an event
> 
> **Fix** was: event('formBuilder.overrideForm', [$this->form]); became event('formBuilder.overrideForm', [&$this->form]);
> 
> and Event::listen('formBuilder.overrideForm', function (&$form) with &
> 
> same in other events was: event('formBuilder.formSubmitted', [$this->form]); became: event('formBuilder.formSubmitted', [&$this->form]);
> 
> **Plugin version** 4.0.5
> 
> **Additional context** Problem with new events

@mplodowski Hi, What about this? the reference will not affect the functionality of the event in any way, but it will add the ability to completely redefine the object if necessary.

**Comment by mplodowski** [comment]
Thanks and sorry for long wait!

Resolved in v4.0.6
